geps/gep-1897.md

```diff
@@ -125,7 +126,235 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
 
 ## API
 
-Details deferred until we reach consensus on what we want to do, and why we want to do this.
+To allow the gateway client to know how to connect to the backend pod, when the backend pod has its own
+certificate, we implement a metaresource that is already mentioned as an example in
+[GEP-713: Metaresources and PolicyAttachment](https://gateway-api.sigs.k8s.io/geps/gep-713/), and as a hypothetical in the
+[Policy Attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/#direct-policy-attachment)
+documentation.
+
+This metaresource is named TLSConnectionPolicy.  In this document, because naming is hard, we chose to retain the
+name TLSConnectionPolicy to advertise alignment with a previously discussed naming choice, but a new name may be
+substituted without blocking acceptance of the content of the API change.
+
+The selection of the applicable Gateway API persona is important in the design of this proposal, because each
+persona is assigned a role which handles specific Gateway API resources.  TLSConnectionPolicy is used by the application
+developer Gateway API persona to convey client certificate settings used in the TLS handshake from Gateway to backend,
+because this persona handles resources involved with application access and configuration, such as Routes and Services.
+Choosing any other role would move the application-related responsibility from the application developer role to that
+role, which violates the role-oriented design principle of Gateway API. As mentioned in Non-goal #7, providing a
+mechanism for the cluster operator gateway role to override gateway to backend TLS settings is not covered here, but can
+be addressed in a future update should the need arise.
+
+TLSConnectionPolicy is defined as a Direct Policy Attachment without defaults or overrides, applied to a Service that
+accesses the backend in question, where the TLSConnectionPolicy resides in the same namespace as the Service it is
+applied to.  The TLSConnectionPolicy and the Service must reside in the same namespace in order to prevent the
+complications involved with sharing trust across namespace boundaries.  By choosing the Service resource rather than the
+Route resource, we can reuse the same TLSConnectionPolicy for all the different Routes that might point to this Service.
+For the use case where certificates are stored in their own namespace, users may create Secrets and use ReferenceGrants
```

Comment:
It may be further down, but I feel like we should probably specify whether or not implementations SHOULD or MUST require and/or respect a ReferenceGrant for cross-namespace Secret sharing with respect to TLSConnection policy (i.e. is an implementation that doesn't require a ReferenceGrant for cross-namespace sharing non-conformant).

Reply:
@keithmattix I documented that references to a different namespace are invalid here: https://github.com/kubernetes-sigs/gateway-api/pull/2113/files/5d84379d2aa570958ce6dbd3fb8cf187447e9f95#diff-fbb3fdd16d81d10c31ffcac7b57c3107034f697fcf1c7b291f5d9e7ec51a4125R225-R226

Do you mean something else?  Or do I need to make TLSConnectionPolicy a namespaced resource?

Reply:
Sorry for the delayed response; either one of your comments addresses my concern. I'll defer to others like @robscott on the partial requirement of ReferenceGrant

Reply:
This is an interesting conversation and at least tangentially related to my other comments. Do we expect there to be a different CA cert per application? If so, I can't imagine that cross-namespace references would be useful for information like this that is not intended to be secret.

Or do we expect there to be a small set of CAs that are used for all certs served by backends in a cluster? If that's the case, maybe we shouldn't be configuring these per Service and instead have a broader target for this config.

Maybe there's a third option where whatever certs are served by backends are using a small set of CAs, but not all backends or ports are serving certs and so although we could configure the CAs at a higher level, we'd still need to separately configure which ports on which backends TLS certs should be verified?

In any case, I'm not really seeing the use case for cross-namespace references here, but interested in what others think.

Reply:
Going through other comments, it looks like @sanjaypujare was actually asking similar questions in https://github.com/kubernetes-sigs/gateway-api/pull/2113/files#r1233341363 and I didn't fully understand the request until I ended up writing out something awfully similar myself above.

Reply:
@robscott is there an issue we still need to address here?

Reply:
Nope, I think we're good now.